app/models/ems_refresh/link_inventory.rb

```diff
@@ -14,9 +14,9 @@ def link_ems_inventory(ems, target, prev_relats, new_relats, disconnect = true)
     # Hook up a relationship from the EMS to the root folder
     _log.info("#{log_header} Updating EMS root folder relationship.")
     root_id = new_relats[:ext_management_systems_to_folders][ems.id][0]
-    if root_id.nil?
+    if root_id.nil? && disconnect
       ems.remove_all_children
-    else
+    elsif !root_id.nil?
```

Comment:
How about just `ems.remove_all_children if disconnect` so we can leave the if/else conditions the same?

Reply:
OK